src/core/exec-invoke.c

```diff
@@ -2315,6 +2365,10 @@ static int setup_exec_directory(
 
         FOREACH_ARRAY(i, context->directories[type].items, context->directories[type].n_items) {
                 _cleanup_free_ char *p = NULL, *pp = NULL;
+                int idmapping_supported;
+                uid_t chown_uid = uid;
+                gid_t chown_gid = gid;
+                bool do_chown = false;
```

Comment:
please move these down to where these are first use.

(we used to have a different coding style on this, but for new code, please declare variables where they are first used, in particular in long functions like this one)